### ISSUE TITLE
[match] Getting error when default keychain is not set

### ISSUE THREAD

**Issue by bguidolim** [fastlane/fastlane#11814]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I'm running the Fastlane on Jenkins, in order to use the same Mac for more than one build at time and as I'm using enterprise and company account, I need to set the keychain that I want to use for each build.

To achieve that, I'm using the follow action on my Fastfile:

`setup_jenkins(set_default_keychain: false, add_keychain_to_search_list: true)`

After other stuffs, I'm calling `match`, using `Matchfile` to set my keychain path.

When `match` is trying to install the provisioning profile, I'm getting this error:

<details>
<pre>
[16:35:15]: Installing provisioning profile...
[16:35:15]: Failure to decode /var/folders/r3/xnw05cjx61sb6s4yf46bz6w00000gn/T/d20180209-67857-ambn3w/profiles/appstore/AppStore_com.enfore.pos.mobileprovision. Exit: 1: security: cert import failed: A default keychain could not be found.
security: problem decoding
</pre>
</details>

After some research on Fastlane code, I found the issue, but I don't know how to fix that in a good way. So I would like to introduce to someone that can fix that easily. It's a simple fix, add `-k` parameters with the current keychain path on this line below:

https://github.com/fastlane/fastlane/blob/83f9895f557293862a9f7546e0ff06cd4b3c1440/fastlane_core/lib/fastlane_core/provisioning_profile.rb#L82

As the parameter `-k` is not in use, it's trying to use the default keychain, but as I said, I'm not using a default keychain.

For now I set my Jenkins to execute one build at time to avoid problems, but would be nice to have that fix soon.

Thanks.

**Comment by taquitos** [comment]
@bguidolim fastlane doesn't support being run in parallel right now. So, while we could add this option, I'm afraid that fastlane would behavior rather strangely and/or become flaky under CI.

The only way that it could work in parallel is if you ran it as different users so that all the session data and temp files would be sandboxed by user.

**Comment by bguidolim** [comment]
@taquitos got it. Actually I didn’t know about that. I could run in parallel using a single keychain, but thanks for the information. If it’s not possible I will probably create a sandbox.

Anyways, looks weird for me that mostly calls for “security” command uses “-k” and this option is not available only on that line. With that in mind, is it sounds reasonable fix?

**Comment by taquitos** [comment]
I'm not super familiar with `security` command, but if most calls to it utilize the `-k` option, I think that's totally reasonable to add it!
